### PR TITLE
Remove context package to allow preloaded subincludes to subinclude other rules

### PR DESCRIPTION
**Pull request by Tatskaari**

### CODE REVIEW

File: src/parse/init.go [hunk 1]
```diff
@@ -55,15 +59,49 @@ func newAspParser(state *core.BuildState) *asp.Parser {
...
+
+func (p *aspParser) preloadSubincludes(state *core.BuildState) {
+	includes := state.Config.Parse.PreloadSubincludes
+	if state.RepoConfig != nil {
+		// TODO(jpoole): is this the right thing to do?
+		includes = append(includes, state.RepoConfig.Parse.PreloadSubincludes...)
+	}
+	for _, inc := range includes {
+		if inc.IsPseudoTarget() {
+			log.Fatalf("Can't preload psudotarget %v", inc)
```
Comment (samwestmoreland) [inline src/parse/init.go:81]: nit: pseudotarget

File: src/parse/init.go [hunk 2]
```diff
@@ -55,15 +59,49 @@ func newAspParser(state *core.BuildState) *asp.Parser {
...
+func (p *aspParser) WaitForInit() {
+	<-p.init
+}
+
+func (p *aspParser) preloadSubincludes(state *core.BuildState) {
+	includes := state.Config.Parse.PreloadSubincludes
+	if state.RepoConfig != nil {
+		// TODO(jpoole): is this the right thing to do?
+		includes = append(includes, state.RepoConfig.Parse.PreloadSubincludes...)
+	}
```
Comment (samwestmoreland) [inline src/parse/init.go:78]: Just trying to get my head around this - can we have both a Config and a RepoConfig on a state? Or does having a RepoConfig imply that the state belongs to a subrepo and so there's no Config?

**Review by samwestmoreland — APPROVED** [review]
Nice! Also looking good in core3 with these changes and the python plugin rules.

File: rules/subrepo_rules.build_defs [hunk 3]
```diff
@@ -327,3 +327,11 @@ def plugin_repo(name:str, revision:str,  plugin:str="", owner:str="please-build"
         ]),
         _subrepo = True,
     )
+
+    subrepo(
```
[the remark this answers is not in the document]
Comment (peterebden) [reply rules/subrepo_rules.build_defs:331]: you have done a good

**Review by peterebden — APPROVED** [review]
Looks good; there's quite a bit going on but seems reasonable. Nice to see contextPkg go, I never liked that!

**Pull request merged**